File: Languages/Ruby/Day016.rb
# Languages - Ruby - Day 16
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that checks if a number is an Armstrong Number. An Armstrong Number is a number that is equal to the sum of its digits each raised to the power of the number of digits.

# Function to check if a number is an Armstrong Number
def is_armstrong(n)
  return false unless n.is_a?(Integer) && n >= 0

  n == n.to_s.chars.map(&:to_i).map { |x| x ** n.to_s.size }.sum
end

# Function to prompt user input
def prompt_user
  print "Enter a positive integer to check if it's an Armstrong Number: "
  input = gets.chomp

  if input.match?(/^\d+$/)
    number = input.to_i
    if is_armstrong(number)
      puts "#{number} is an Armstrong Number."
    else
      puts "#{number} is not an Armstrong Number."
    end
  else
    puts "Invalid input. Please enter a positive integer."
  end
end

# Run the prompt
prompt_user

File: Languages/Rust/Day016.rs
// Languages - Rust - Day 16
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that checks if a number is an Armstrong Number. An Armstrong Number is a number that is equal to the sum of its digits each raised to the power of the number of digits.

use std::io;

// Function to check if a number is an Armstrong Number
fn is_armstrong_number(num: i32) -> bool {
    if num < 0 {
        println!("Please enter a positive integer.");
        return false;
    }

    let num_str = num.to_string();
    let num_len = num_str.len() as u32;
    let mut sum = 0;

    for c in num_str.chars() {
        if let Some(digit) = c.to_digit(10) {
            sum += digit.pow(num_len);
        } else {
            println!("Invalid character encountered.");
            return false;
        }
    }

    sum == num as u32
}

fn main() {
    println!("Enter a positive integer to check if it's an Armstrong Number:");

    let mut input = String::new();

    match io::stdin().read_line(&mut input) {
        Ok(_) => {
            let input = input.trim();

            match input.parse::<i32>() {
                Ok(num) if num >= 0 => {
                    if is_armstrong_number(num) {
                        println!("{} is an Armstrong Number", num);
                    } else {
                        println!("{} is not an Armstrong Number", num);
                    }
                }
                _ => println!("Invalid input. Please enter a positive integer."),
            }
        }
        Err(_) => println!("Failed to read input."),
    }
}
